examples/decomposition/apply_covid.py

```diff
@@ -0,0 +1,180 @@
+"""
+Apply CP decomposition to COVID-19 Serology Dataset
+"""
+
+##############################################################################
+# Introduction
+# -----------------------
+# PARAFAC (CP) decomposition is extremely useful in dimensionality reduction, allowing us
+# to develop models that are both representative and compact while retaining crucial patterns
+# between subjects. Here, we provide an example of how it can be applied to biomedical research.
+#
+# Systems serology is a new technology that examines the antibodies from a patient's serum, aiming
+# to comprehensively profile the interactions between the antibodies and
+# [Fc receptors](https://en.wikipedia.org/wiki/Fc_receptor) alongside other types of immunological
+# and demographic data. Here, we will apply CP decomposition to a [COVID-19 system serology dataset]
+# (https://www.sciencedirect.com/science/article/pii/S0092867420314598).
+#
+# Details of this analysis as well as more in-depth biological implication can be found in
+# [this work](https://www.embopress.org/doi/full/10.15252/msb.202110243).
+#
+# We first import this dataset of a panel of COVID-19 patients:
+
+import numpy as np
+import pandas as pd
+from tensorly.datasets.imports import COVID19_data
+
+data = COVID19_data()
+
+##############################################################################
+# Apply CP decomposition to this dataset with Tensorly
+# -----------------------
+# Now we apply CP decomposition to this dataset.
+
+import tensorly as tl
+from tensorly.decomposition import parafac
+
+comps = np.arange(1, 7)
+CMTFfacs = [parafac(data.tensor, cc, tol=1e-10, n_iter_max=1000,
+                    linesearch=True, orthogonalise=2) for cc in comps]
+
+##############################################################################
+# To evaluate how well CP decomposition explains the variance in the dataset, we plot the percent
+# variance reconstructed (R2X) for a range of ranks.
+
+def calcR2X(tFac, tIn=None):
```

Comment:
Would it be worth calling it reconstructed_variance and adding a very short docstring?
Is it used enough to add to the metrics @aarmey @cohenjer ?
In it's current form it would not be ideal due to depending on NumPy but we could probably adapt. Anyway, this is an issue separate from this PR :)

Reply:
@JeanKossaifi I've been meaning to bring up whether we should include this. :-) We use it a *ton* so it would be great to add. We'll do so in a separate PR.